dev/tests/scheduler.rs

```diff
@@ -452,3 +452,49 @@ async fn schedule_request_lock() {
     assert_eq!(drone1, drone2);
     assert_eq!(backend1, backend2);
 }
+
+
+#[integration_test]
+async fn schedule_request_lock_with_bearer_token() {
+    let nats = Nats::new().await.unwrap();
+    let nats_conn = nats.connection().await.unwrap();
+    let state = start_state_loop(nats_conn.clone()).await.unwrap();
+    let _scheduler_guard = expect_to_stay_alive(run_scheduler(nats_conn.clone(), state));
+    let drone_id = DroneId::new_random();
+    let mock_agent = MockAgent::new(nats_conn.clone(), &drone_id).await;
+    sleep(Duration::from_millis(100)).await;
+
+    nats_conn
+        .publish(&DroneStatusMessage {
+            cluster: ClusterName::new("plane.test"),
+            drone_id: drone_id.clone(),
+            drone_version: PLANE_VERSION.to_string(),
+            ready: true,
+            state: DroneState::Ready,
+            running_backends: None,
+        })
+        .await
+        .unwrap();
+
+    sleep(Duration::from_millis(100)).await;
```

Comment:
what condition is this sleep to ensure?

Reply:
It was in the test I copied, I've forgotten at this point. I agree that we should go back and remove sleeps, but I'd rather merge with some sleeps in test for now and then clean up later.